samcli/lib/logs/formatter.py

```diff
@@ -0,0 +1,220 @@
+
+        Returns
+        -------
+        iterable of string
+            Iterable that returns a formatted event as a string.
+        """
+
+        for operation in self.formatter_chain:
+
+            # Make sure the operation has access to certain basic objects like colored
+            partial_op = functools.partial(operation, colored=self.colored)
+            event_iterable = map(partial_op, event_iterable)
+
+        return event_iterable
+
+    @staticmethod
+    def _pretty_print_event(event, colored):
+        """
+        Basic formatter to convert an event object to string
+
+        Parameters
+        ----------
+        event
+        colored
+
+        Returns
+        -------
+
+        """
+        event.timestamp = colored.yellow(event.timestamp)
+        event.log_stream_name = colored.cyan(event.log_stream_name)
+
+        return ' '.join([event.log_stream_name, event.timestamp, event.message])
+
+
+class LambdaLogMsgFormatters(object):
+    """
+    Format logs printed by AWS Lambda functions.
+
+    This class is a collection of static methods that can be used within a formatter chain.
+    """
+
+    @staticmethod
+    def colorize_errors(event, colored):
+        """
+
+        Parameters
+        ----------
+        message
+
+        Returns
+        -------
+
+        """
+
+        nodejs_crash_msg = "Process exited before completing request"
+        timeout_msg = "Task timed out"
+
+        if nodejs_crash_msg in event.message \
+                or timeout_msg in event.message:
+            event.message = colored.red(event.message)
+
+        return event
+
+    @staticmethod
+    def colorize_reports(event, colored):
+        """
+
+        Parameters
+        ----------
+        message
+
+        Returns
+        -------
+
+        """
+
+        if event.message.startswith("START") or event.message.startswith("END") or event.message.startswith("REPORT"):
+            event.message = colored.white(event.message)
+
+        return event
+
+
+class KeywordHighlighter(object):
+
+    def __init__(self, keyword=None):
+        self.keyword = keyword
+
+    def highlight_keywords(self, event, colored):
+        if self.keyword:
+            highlight = colored.underline(self.keyword)
+            event.message = event.message.replace(self.keyword, highlight)
+
+        return event
+
+
+class JSONMsgFormatter(object):
+    """
+    Pretty print JSONs within a message
+    """
+
+    @staticmethod
+    def format_json(event, colored):
+        """
+        If the event message is a JSON string, then pretty print the JSON with 2 indents.
+
+        Parameters
+        ----------
+        event
+        colored
+
+        Returns
+        -------
+
+        """
+
+        try:
+            if event.message.startswith("{"):
+                msg_dict = json.loads(event.message)
+                event.message = json.dumps(msg_dict, indent=2)
```

Comment:
In Py2 this dictionary will be different than Py3. For consistency, can we use `OrderedDict` instead? This will make Py2 and Py3 output the same.

You can do this in the `loads` call by:
`json.loads(event.message, object_pairs_hook=OrderedDict)`

Reply:
I updated to use `sort_keys=True` so we pretty print & sort the keys to simplify visually inspecting JSON data. This will break the original ordering anyway.